src/OrleansRuntime/MembershipService/GrainBasedMembershipTable.cs

```diff
@@ -32,6 +34,16 @@ internal class GrainBasedMembershipTable : Grain, IMembershipTable
         private InMemoryMembershipTable table;
         private TraceLogger logger;
 
+        public Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
+        {
+            throw new InvalidOperationException("This method should not be called directly");
```

Comment:
No, I think throwing is wrong here. It will break the common code in tests and all the usage of GrainBasedMembershipTable , since we are calling InitializeMembershipTable for all implementatiions.
I think it should be no-op instead.